libndt.hpp

```diff
@@ -1960,6 +2017,20 @@ bool Client::ndt7_upload() noexcept {
     constexpr auto measurement_interval = 0.25;
     std::chrono::duration<double> interval = now - latest;
     if (interval.count() > measurement_interval) {
+#ifdef __linux__
+      // Read tcp_info data for the socket and print it as JSON.
+      struct tcp_info tcpinfo{};
+      socklen_t soerrlen = sizeof(tcpinfo);
+      if (sys_getsockopt(sock_, IPPROTO_TCP, TCP_INFO, (void *)&tcpinfo,
+                             &soerrlen) == 0) {
+        nlohmann::json doc;
+        doc["TCPInfo"] = nlohmann::json();
+#define XX(lower_, upper_) doc["TCPInfo"][#upper_] = (uint64_t) tcpinfo.lower_;
+        NDT7_ENUM_TCP_INFO
+#undef XX
+        on_result("ndt7", "upload", doc.dump());
```

Comment:
In addition to delivering this information to the client, we should also send it to the server

Reply:
Done. Thoughts?